### DISCUSSION TITLE
Exisiting APIs custom changes

### DISCUSSION THREAD

**Discussion by TheLast-King** [medusajs/medusa#6497] [Q&A]
I would like to know( i have installed medusa-backend cli) for admin api:
i) How can I make changes in existing route of products of store and admin API.
ii) How can I add more api routes with custom functionality in exisiting apis like products/regions etc.

**Comment by olivermrbl** [comment]
> i) How can I make changes in existing route of products of store and admin API.

Currently, you cannot, and you should not. In Medusa 2.0 releasing later this year, we will introduce Workflows across all API endpoints. These workflows will come with _hooks_, that allow you to modify the behavior of our pre-built flow of events by injecting additional steps or removing existing ones. You can read more about it [here](https://docs.medusajs.com/development/workflows/).

**Reply by vizzv** [reply]
> > i) How can I make changes in existing route of products of store and admin API.
> 
> Currently, you cannot, and you should not. In Medusa 2.0 releasing later this year, we will introduce Workflows across all API endpoints. These workflows will come with _hooks_, that allow you to modify the behavior of our pre-built flow of events by injecting additional steps or removing existing ones. You can read more about it [here](https://docs.medusajs.com/development/workflows/).

https://docs.medusajs.com/development/workflows link gives Page Not Found .

**Comment by olivermrbl** [comment]
> ii) How can I add more api routes with custom functionality in exisiting apis like products/regions etc.

You should use our [API Routes](https://docs.medusajs.com/development/api-routes/create) to achieve this.
